### PR TITLE
[#89] Initial approach to the integration tests

### PR DESCRIPTION
**Pull request by marcingrzejszczak**
Using docker-compose in the `pre-integration-test` phase. In that phase it's also booting the app from the jar (we're checking that it's package properly). 

Then in the `integration-test` phase it's executing 2 tests. 

Finally in the `post-integration-test` phase (that's always executed regardless of the `integration-test` phase status) it's killing docker containers, and killing the started apps. Also it's printing 500 lines of logs from the started app (for debugging).

### CODE REVIEW

**Comment by marcingrzejszczak** [comment]
CC @dsyer , @spencergibb , @adriancole

File: .travis.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ before_install:
   - echo "https://$GH_TOKEN:@github.com" > .git/credentials
   - gem install asciidoctor
 install:
-- ./mvnw install -P docs -q -U -DskipTests=true -Dmaven.test.redirectTestOutputToFile=true
+- ./mvnw install -P docs -q -U -DskipTests=true -Dexec.skip=true -Dmaven.test.redirectTestOutputToFile=true
```
Comment (adriancole) [inline .travis.yml]: add a comment above about exec.skip? it is a rarely used option

Comment (adriancole) [inline .travis.yml]: ex `# don't run integration tests by default`

Comment (marcingrzejszczak) [inline .travis.yml]: I would but ATM it's not working :) I officialy hate Maven

**Comment by adriancole** [comment]
summary of thoughts offline. would be great if the lifecycle of the components could be managed by junit somehow. they'd be easier to run I think

**Pull request closed without merging**